### PR TITLE
#trivial fix button and button tests eslint issues

### PR DESCRIPTION
**Pull request by igorbrasileiro**
#### What is the purpose of this pull request?
as the title says. #1082  

#### Types of changes
- [x] Bugfix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/components/Button/index.tsx [hunk 1]
```diff
@@ -41,7 +41,7 @@ const propTypes = {
   /** (Button spec attribute) */
   autoComplete: PropTypes.string,
   /** (Button spec attribute) */
-  disabled: PropTypes.bool,
+  disabled: PropTypes.bool.isRequired,
```
Comment (emersonlaurentino) [inline react/components/Button/index.tsx]: when they use the button, is it necessary to change it to prop disabled?

File: react/components/Button/index.tsx [hunk 2]
```diff
@@ -27,7 +27,7 @@ const propTypes = {
   /** Loading state */
   isLoading: PropTypes.bool,
   /** [DEPRECATED] If you are using just an Icon component inside, use this as true */
-  icon: PropTypes.bool,
+  icon: PropTypes.bool.isRequired,
```
Comment (emersonlaurentino) [inline react/components/Button/index.tsx]: same with icon?

File: react/components/Button/index.tsx (hunk 1 above)
Comment (igorbrasileiro) [reply react/components/Button/index.tsx]: Nope, but disabled and icon is by default props false. If this props isn't required, eslint will warn to fix this [line](https://github.com/vtex/styleguide/blob/375d01193da371167c2e5a861024bdab43eb2f6f/react/components/Button/index.tsx#L162). The eslint fix changes from `||` to `??` operators, but these two operators aren't equivalents.

Comment (emersonlaurentino) [reply react/components/Button/index.tsx]: but it is not obliging the user to pass the `disabled` and `icon` props to the button that will resolve this.

Comment (igorbrasileiro) [reply react/components/Button/index.tsx]: Ok, so can I remove it from default props?

File: react/components/Button/index.tsx [hunk 3]
```diff
@@ -159,8 +159,8 @@ const Button: FC<Props> = props => {
     }
   }
 
-  const disabled = props.disabled || isLoading
-  const iconOnly = icon || props.iconOnly
+  const disabled = (props.disabled as boolean) || isLoading
+  const iconOnly = (icon as boolean) || props.iconOnly
```
Comment (emersonlaurentino) [inline react/components/Button/index.tsx]:
```suggestion
  const disabled = !!props.disabled || isLoading
  const iconOnly = !!icon || props.iconOnly
```

Comment (igorbrasileiro) [reply react/components/Button/index.tsx]: fixed

File: react/components/Button/index.tsx [hunk 4]
```diff
@@ -159,7 +159,9 @@ const Button: FC<Props> = props => {
     }
   }
 
+  // eslint-disable-next-line
   const disabled = props.disabled || isLoading
+  // eslint-disable-next-line
```
Comment (emersonlaurentino) [inline react/components/Button/index.tsx:164]:
```suggestion
  // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
```

File: react/components/Button/index.tsx [hunk 5]
```diff
@@ -159,7 +159,9 @@ const Button: FC<Props> = props => {
     }
   }
 
+  // eslint-disable-next-line
```
Comment (emersonlaurentino) [inline react/components/Button/index.tsx:162]:
```suggestion
  // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
```

File: react/components/Button/index.tsx [hunk 6]
```diff
@@ -159,7 +159,9 @@ const Button: FC<Props> = props => {
     }
   }
 
+  // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
```
Comment (matheusps) [inline react/components/Button/index.tsx:162]: don't disable

Comment (emersonlaurentino) [reply react/components/Button/index.tsx:162]: why?

**Pull request merged**